### PR TITLE
feat: check if invoice is paid

### PR DESCRIPTION
**Pull request by dylancom**
Should solve #289

### CODE REVIEW

File: src/app/screens/Receive.tsx [hunk 1]
```diff
@@ -32,6 +34,19 @@ function Receive() {
     });
   }
 
+  useEffect(() => {
+    if (invoice) {
+      const pollForPayment = poll({
+        fn: () => utils.call("checkPayment", { paymentHash: invoice.rHash }),
+        validate: (payment) => payment.paid,
+        interval: 5000,
+        maxAttempts: 10,
```
Comment (bumi) [inline src/app/screens/Receive.tsx:43]: I think we should poll a little longer but decrease the interval a bit (maybe 3 seconds? - 5 seconds feels quiet long?)

File: src/extension/background-script/connectors/lnd.ts [hunk 2]
```diff
@@ -63,6 +65,17 @@ class Lnd implements Connector {
     });
   }
 
+  async checkPayment(args: CheckPaymentArgs): Promise<CheckPaymentResponse> {
```
Comment (bumi) [inline src/extension/background-script/connectors/lnd.ts:68]: to be more generally used ideally we return more invoice data and maybe even call the function `getInvoice`? 
sadly it seems lndhub/lnbits to not really return much useful invoice data.

**Comment by bumi** [comment]
I heard some reports that it people had been waiting for a payment and had been confused. good that you fix it with this PR!

**Comment by RicardoCapricciosa** [comment]
How exactly can I test this? I installed the Firefox extension from above and sent some sats to may mobile wallet. I've received this notification from firefox 
### image

The text is a bit confusing, because I sent the 500 sats FROM Alby, not TO Alby

**Comment by RicardoCapricciosa** [comment]
> How exactly can I test this? I installed the Firefox extension from above and sent some sats to my mobile wallet. I've received this notification from firefox
> ### image
> 
> The text is a bit confusing, because I sent the 500 sats FROM Alby, not TO Alby

Ok, I just tried again with the released version, which behaved the same. So the above assumption wasn't correct :-)

**Comment by RicardoCapricciosa** [comment]
ok, after looking at #289 I got it
image
image

So it works. But what I find a bit confusing is, that once the Alby window is closed the invoice is gone. At first I had a couple of failed attempts to pay the invoice, but that was a problem on my end (zeus, tor, etc.). But I could not try to pay with another wallet because the invoice was gone although not yet paid. I had to create a new invoice.

File: src/app/screens/Receive.tsx [hunk 3]
```diff
@@ -75,6 +101,21 @@ function Receive() {
...
+              <span>waiting for payment...</span>
+            </div>
+          )}
+          {!paid && !pollingForPayment && (
+            <Button
+              onClick={() => checkPayment(invoice.rHash)}
+              label="Check payment status"
+            />
+          )}
+          {paid && <p>Payment received!</p>}
```
Comment (bumi) [inline src/app/screens/Receive.tsx:117]: I think we should then also hide the QR code or maybe have some kind of ✅ overlay. 
or maybe @johnsBeharry confetti?

Comment (bumi) [reply src/app/screens/Receive.tsx:117]: is this too much of a dependency?

**Comment by bumi** [comment]
@RicardoCapricciosa nice, thanks for testing. those are some good points. 

can you create an issue for those two topics? I think we do not have issues for those yet. 
The reload is caused because whenever the extension is opened the popup is reloaded and we do not store state across those reloads - which I think for now is fine, it is too complex for now to change that. 

The confusion with "To Alby" is more important.

File: src/app/screens/Receive.tsx (hunk 3 above)
Comment (dylancom) [reply src/app/screens/Receive.tsx:117]: Agree, yeah I like the "fun-factor" of that and think it's a small lib to add :)

Comment (bumi) [reply src/app/screens/Receive.tsx:117]: yeah, we need more fun factors! :)

Comment (RicardoCapricciosa) [reply src/app/screens/Receive.tsx:117]: > Agree, yeah I like the "fun-factor" of that and think it's a small lib to add :)

I also agree, confetti are nice, at least in the beginning. Once I use the app five times a day the confetti animation might get a bit old, but for now I think it's great!

**Comment by bumi** [comment]
looks nice. 
Though I still think we should replace or overlay the QR code to make it visual that it is paid. 
the `Payment received!` message can easily be missed I think.

**Comment by dylancom** [comment]
Updated the success screen. Could improve it in the future with for e.g. a modal sliding up from the bottom. Something like: 
image

**Comment by RicardoCapricciosa** [comment]
> @RicardoCapricciosa nice, thanks for testing. those are some good points.
> 
> can you create an issue for those two topics? I think we do not have issues for those yet. The reload is caused because whenever the extension is opened the popup is reloaded and we do not store state across those reloads - which I think for now is fine, it is too complex for now to change that.
> 
> The confusion with "To Alby" is more important.

- notification confusion text: https://github.com/getAlby/lightning-browser-extension/issues/473#issue-1084187974
- Alby state storing: https://github.com/getAlby/lightning-browser-extension/issues/474#issue-1084190298

**Comment by bumi** [comment]
perfect thanks @RicardoCapricciosa

**Review by bumi — APPROVED** [review]

**Pull request merged**